README.md

```diff
@@ -1,2 +1,366 @@
+[<img align="right" title="Share on Twitter" src="https://docs.cloudposse.com/images/ionicons/social-twitter-outline-2.0.1-16x16-999999.svg" />][share_twitter]
+
+
+
+
+It's 100% Open Source and licensed under the [APACHE2](LICENSE).
+
+
+
+
+
+
+
+
+
+
+
+
+## Introduction
+
+
+If you answer "yes" to any of these questions, then look no further!
+
+* Have you ever wished you could easily filter sensitive output from a `terraform plan` or `terraform apply`?
+* Do you use terraform providers that leak sensitive data to `stdout` (e.g. `terraform-github-provider`)?
+
+**Yes?** Great! Then this utility is for you.
+
+The `tfmask` utility will replace the "old value" and the "new value" with the masking character (e.g. `*`).
+  
+__NOTE__: `tfmask` will preserve the name of the nodes in the graph
+
+## Usage
+
+
+__NOTE__: The utility supports a number of configuration settings which can be passed via environment variables.
+
+| Environment Variable | Description                                  | Default    |
+|----------------------|----------------------------------------------|------------| 
+| `TFMASK_CHAR`        |  Character used to mask all output           | `*`        | 
+| `TFMASK_REGEX`       | Regular expression used to match graph nodes | [see code] | 
+  
+The basic usage looks like this. We're going to run `terraform plan` and filter it through `tfmask`:
+
+```sh
+terraform plan | tfmask
+```
+
+### Direnv
+
+You can use `tfmask` with [`direnv`](https://direnv.net/) to set the defaults.
+
+Example `.envrc`:
+
+```sh
+# Export terraform environment
+export TFMASK_CHAR="#"
+export TFMASK_REGEX="(?i)^.*(secret|password|oauth|token|key).*$"
+```
+
+<details>
+  <summary>Example of Masked Output</summary>
+
+  ```sh
+  Terraform will perform the following actions:
+
+    ~ module.atlantis.module.web_app.module.ecs_codepipeline.aws_codepipeline.source_build_deploy
+        stage.0.action.0.configuration.%:          "4" => "5"
+        stage.0.action.0.configuration.OAuthToken: "" => "*******************************************"
+
+  Plan: 0 to add, 1 to change, 0 to destroy.
+  ```
+</details>
+
+
+
+
+## Examples
+
+
+### Compiling the Binary
+
+```sh
+make go/build
+```
+
+ ### Use with Terraform Plan
+
+ Many terraform providers unintentionally leak sensitive information when running `terraform plan`. **This is very bad.**
+
+<details>
+  <summary>Example of a Leaked Secret from a Terraform Plan</summary>
+
+  ```sh
+  An execution plan has been generated and is shown below.
+  Resource actions are indicated with the following symbols:
+    ~ update in-place
+
+  Terraform will perform the following actions:
+
+    ~ module.example.aws_codepipeline.source_build_deploy
+        stage.0.action.0.configuration.%:          "4" => "5"
+        stage.0.action.0.configuration.OAuthToken: "" => "efba05dbe9b94ba18ae3737a6d6de16eefba05dbe9b9"
```

Comment:
Has this been revoked?

Reply:
Ah, I've read below.

Reply:
Haha, yea, good looking out =) 

Yea, was just some random string I generated from a mangled uuid without dashes.